src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+    """
+    SatelliteScheduleProblem is a synthetic scheduling problem in which a
+    number of vehicles must be assigned to view as many requests in a
+    2-dimensional plane as possible. Each vehicle moves horizontally across
+    the plane, has minimum and maximum view angle, and has a maximum rotation
+    rate (i.e. the rate at which the vehicle can reorient vertically from one
+    target to the next).
+
+    The problem is represented as an infeasibility graph and can be solved by
+    finding the Maximum Independent Set.
+    """
+
+    def __init__(
+            self,
+            num_satellites: int = 6,
+            view_height: float = 0.25,
+            view_coords: Optional[np.ndarray] = None,
+            num_requests: int = 48,
+            requests: Optional[np.ndarray] = None,
+            turn_rate: float = 2,
+            solution_criteria: float = 0.99,
+            seed: int = 42,
+    ):
+        """ Create a SatelliteScheduleProblem.
+        Parameters
+        ----------
+        num_satellites : int, default = 6
+            The number of satellites to generate schedules for.
+        view_height : float, default = 0.25
+            The range from minimum to maximum viewable angle for each satellite.
+        view_coords : Optional[np.ndarray], default = None
+            The view coordinates (i.e. minimum viewable angle) for each
+            satellite in a numpy array. If None, view coordinates will be
+            evenly distributed across the viewable range.
+        num_requests : int, default = 48
+            The number of requests to generate.
+        turn_rate : float, default = 2
+            How quickly each satellite may reorient its view angle.
+        solution_criteria : float, default = 0.99
+            The target for a successful solution. The solver will stop
+            looking for a better schedule if the specified fraction of
+            requests is satisfied.
+        """
+        super(SatelliteScheduleProblem,
+              self).__init__(num_agents=num_satellites,
+                             num_tasks=num_requests,
+                             sat_cutoff=solution_criteria,
+                             seed=seed)
+        self.num_satellites = self.num_agents
+        self.num_requests = self.num_tasks
+
+        self.view_height = view_height
+        if view_coords is None:
+            self.view_coords = np.linspace(0 - view_height / 2,
+                                           1 - view_height / 2,
+                                           num_satellites)
+        else:
+            self.view_coords = view_coords
+        self.agent_attrs = list(zip([view_height] * num_satellites,
+                                    self.view_coords))
+        self.satellites = self.agent_ids
+        self.turn_rate = turn_rate
+        self.requests = None
+        self.qubo_problem = None
+        self.generate_requests(requests)
+
+    def generate_requests(self, requests=None):
```

Comment:
In general, you may want to add typing on what these methods return. Here, e.g. "-> None".

Reply:
Fair point.